src/interfaces.jl

```diff
@@ -110,3 +122,27 @@ function reverse_time(sol::Solver)
     options.stopevent = (t,y)->stopevent(2*t0-t,y)
     return solve(ode_reversed,stepper,options)
 end
+
+
+
+# The the elements of tspan should basically be scalars and support
+# most of the scalar operations.  In particular the element type
+# should be closed under the division.
+function make_consistent{S<:AbstractStepper}(y0, tspan::AbstractVector, stepper::Type{S})
+    t_test = 1/(tspan[end]-tspan[1])
+    Tt = typeof(t_test)
+    t_new = convert(Vector{Tt},tspan)
+
+    y_test = y0./t_test
+    Ty = typeof(y_test)
+    if typejoin(Ty,AbstractArray) == AbstractArray
+        Ey = promote_type(map(typeof,y_test)...)
```

Comment:
On my computer, if `y_test` is a 10^5 element vector (say from a PDE), this takes ~23 seconds.  

I think it would make sense to check that `isleaftype(eltype(y0))` and if false issue a warning: "The `eltype(y0)=$(eltype(y0)` is not a concrete type, this degrades performance substantially."

So
```
    y_test = y0./t_test
    Ty = typeof(y_test)
    if Ty<:AbstractVector
        # if eltype(y0) is non-concrete, then eltype(y_test) will also be non-concrete
        Ey = eltype(y_test)
        if !isleaftype(Ey)
            warn("The `eltype(y0)=$Ey` is not a concrete type, this degrades performance substantially.")
        end
        y_new = copy!(similar(y0,Ey), y0)
    else
        y_new = convert(Ty,y0)
    end
```